### PR TITLE
Added information about limitSecretReferences

### PR DESCRIPTION
**Pull request by soltysh**
@liggitt @deads2k ptal
@adellape ptal once this goes through dev review

### CODE REVIEW

File: admin_guide/service_accounts.adoc [hunk 1]
```diff
@@ -119,3 +119,15 @@ policyConfig:
   openshiftInfrastructureNamespace: openshift-infra
 ----
 ====
+
+To controls whether or not to allow a service account to reference any secret
```
Comment (deads2k) [inline admin_guide/service_accounts.adoc]: To control

File: dev_guide/builds.adoc [hunk 2]
```diff
@@ -1064,6 +1063,9 @@ $ oc secrets add serviceaccount/builder secrets/scmsecret
 ----
 ====
 
+NOTE: Depending on the `*limitSecretReferences*`, `*serviceAccountConfig*` section
```
Comment (deads2k) [inline dev_guide/builds.adoc]: There's no way to know whether or not this is enabled client side.  Is it really worth calling it out as optional?

File: admin_guide/service_accounts.adoc (hunk 1 above)
Comment (deads2k) [inline admin_guide/service_accounts.adoc]: Maybe more prescriptive? 

Set this value to true to require pod secret references to be whitelisted by their service accounts.  Set this value to false to allow pods to reference any secret in the namespace.

**Comment by deads2k** [comment]
It's accurate.

File: dev_guide/builds.adoc (hunk 2 above)
Comment (soltysh) [inline dev_guide/builds.adoc]: Yeah, after I dig into the topic I started feeling the same way. But on the other hand if you go through the steps they seem so artificial. One other though would be to explain this step, why this add is needed, would that make sense?

Comment (deads2k) [inline dev_guide/builds.adoc]: > One other option would be to explain this step, why this add is needed, would that make sense?

Yes, I'd like that better.

File: admin_guide/service_accounts.adoc (hunk 1 above)
Comment (soltysh) [inline admin_guide/service_accounts.adoc]: Done.

**Comment by soltysh** [comment]
@deads2k can you take a look again?

File: admin_guide/service_accounts.adoc [hunk 3]
```diff
@@ -119,3 +119,15 @@ policyConfig:
   openshiftInfrastructureNamespace: openshift-infra
 ----
 ====
+
+Set `limitSecretReferences` field in link:../admin_guide/master_node_configuration.html#master-configuration-files[master configuration]
+file to `true` to require pod secret references, to be whitelisted by their service accounts.
```
Comment (deads2k) [inline admin_guide/service_accounts.adoc]: no comma

**Comment by deads2k** [comment]
nit.

lgtm

**Comment by soltysh** [comment]
Nit fixed.

**Pull request merged**